### PR TITLE
Problem: (fix #1949) non-deterministic test-vectors with upstream schnorrsig

### LINKED ISSUE #1949: Problem: non-deterministic test-vectors with upstream schnorrsig
https://github.com/crypto-com/chain/blob/master/dev-utils/src/commands/test_vector_command.rs#L264
the auxiliary randomness can be fixed and reported as in BIP-340 https://github.com/bitcoin/bips/blob/master/bip-0340/test-vectors.csv

### PR DESCRIPTION
**Pull request by linfeng-crypto**
solution: use `schnorr_sign_aux` instead of `schnorr_sign` and allow users input a `aux_payload`

### CODE REVIEW

File: client-common/src/key/private_key.rs [hunk 1]
```diff
@@ -49,6 +53,23 @@ impl PrivateKeyAction for PrivateKey {
         Ok(signature)
     }
 
+    fn schnorr_sign_aux(&self, tx: &Transaction, aux_payload: &[u8]) -> Result<SchnorrSignature> {
```
Comment (tomtau) [inline client-common/src/key/private_key.rs:56]:
```suggestion
    fn schnorr_sign_unsafe(&self, tx: &Transaction, aux_payload: &[u8]) -> Result<SchnorrSignature> {
```

As this is not meant to be called anywhere else in the client and is only used in dev-utils, I suggest the name should reflect it's "dev-only"

File: dev-utils/src/dev_utils.rs [hunk 2]
```diff
@@ -87,11 +96,27 @@ impl DevUtils {
...
+#[cfg(test)]
+mod tests {
+
+    #[test]
+    fn test_abc() {
+        let a: [u8; 32] = [0; 32];
+        let b = hex::encode(a);
+        println!("{}", b);
+    }
+}
```
Comment (tomtau) [inline dev-utils/src/dev_utils.rs:122]: ?

Comment (linfeng-crypto) [reply dev-utils/src/dev_utils.rs:122]: I'll remove it.

**Pull request merged**